plugin/trino-loki/src/main/java/io/trino/plugin/loki/LokiRecordCursor.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.loki;
+
+import com.google.common.collect.ImmutableList;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+import io.trino.spi.block.SqlMap;
+import io.trino.spi.connector.RecordCursor;
+import io.trino.spi.type.MapType;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.TypeUtils;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.trino.spi.block.MapValueBuilder.buildMapValue;
+import static io.trino.spi.type.DoubleType.DOUBLE;
+import static io.trino.spi.type.VarcharType.createUnboundedVarcharType;
+import static java.util.Objects.requireNonNull;
+
+public class LokiRecordCursor
+        implements RecordCursor
+{
+    final List<LokiColumnHandle> columnHandles;
+    protected final int[] fieldToColumnIndex;
+    final LokiQueryResultIterator resultIterator;
+
+    public LokiRecordCursor(List<LokiColumnHandle> columnHandles, LokiQueryResultIterator queryResult)
+    {
+        this.columnHandles = ImmutableList.copyOf(columnHandles);
+        this.resultIterator = queryResult;
+
+        fieldToColumnIndex = new int[columnHandles.size()];
+        for (int i = 0; i < columnHandles.size(); i++) {
+            LokiColumnHandle columnHandle = columnHandles.get(i);
+            fieldToColumnIndex[i] = columnHandle.ordinalPosition();
+        }
+    }
+
+    @Override
+    public boolean advanceNextPosition()
+    {
+        return this.resultIterator.advanceNextPosition();
+    }
+
+    @Override
+    public long getCompletedBytes()
+    {
+        return 0;
+    }
+
+    @Override
+    public long getReadTimeNanos()
+    {
+        return 0;
+    }
+
+    @Override
+    public Type getType(int field)
+    {
+        checkArgument(field < columnHandles.size(), "Invalid field index");
+        return columnHandles.get(field).type();
+    }
+
+    Object getFieldValue(int field)
+    {
+        // TODO: checkState(current != null, "Cursor has not been advanced yet");
+
+        int columnIndex = fieldToColumnIndex[field];
+        return switch (columnIndex) {
+            case 0 -> getSqlMapFromLabels(this.columnHandles.get(columnIndex).type(), this.resultIterator.getLabels());
```

Comment:
Avoid magic numbers. This relies on the order of columns defined in `LokiMetadata`. Instead of specifying the ordinal position there, index column handle names and use them in this switch. They must be unique anyway.

Reply:
True. FWIW this was taken from the [Prometheus connector](https://github.com/trinodb/trino/blob/b57552d2d085f81ce92b95636b71676917a95f57/plugin/trino-prometheus/src/main/java/io/trino/plugin/prometheus/PrometheusRecordCursor.java#L124-L130).